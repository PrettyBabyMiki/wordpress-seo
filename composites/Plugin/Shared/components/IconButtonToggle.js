import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

import colors from "../../../../style-guide/colors.json";
import { Icon } from "./Icon";
import { rgba } from "../../../../style-guide/helpers";

const ChangingIconButtonBase = styled.button`
	box-sizing: border-box;
	min-width: 32px;
	display: inline-block;
	border: 1px solid ${ colors.$color_button_border };
	background-color: ${ props => props.pressed ? props.pressedBackground : props.unpressedBackground };
	box-shadow: ${ props => props.pressed
		? `inset 0 2px 0 ${ rgba( props.pressedBoxShadowColor, 0.7 ) }`
		: `0 1px 0 ${ rgba( props.unpressedBoxShadowColor, 0.7 ) }` };
	border-radius: 3px;
	cursor: pointer;
	padding: 0;
	height: ${ props => props.pressed ? "23px" : "24px" };

	&:hover {
		border-color: ${ props => props.hoverBorderColor }
	}
`;

/**
 * Returns the ChangingIconButton component.
 *
 * @param {object} props Component props.
 *
 * @returns {ReactElement} ChangingIconButton component.
 */
const ChangingIconButton = ( props ) => {
	return (
		<ChangingIconButtonBase
			onClick={ props.onClick }
			pressed={ props.pressed }
			unpressedBoxShadowColor={ props.unpressedBoxShadowColor }
			pressedBoxShadowColor={ props.pressedBoxShadowColor }
			pressedBackground={ props.pressedBackground }
			unpressedBackground={ props.unpressedBackground }
			id={ props.id }
			aria-label={ props.ariaLabel }
			aria-pressed={ props.pressed }
			unpressedIconColor={ props.unpressedIconColor }
			pressedIconColor={ props.pressedIconColor }
			hoverBorderColor={ props.hoverBorderColor }
		>
			{ props.pressed
				? <Icon
					icon={ props.icon }
					color={ props.pressedIconColor }
					size="18px"/>
				: <Icon
					icon={ props.icon }
					color={ props.unpressedIconColor }
					size="18px"/>
			}
		</ChangingIconButtonBase>
	);
};

ChangingIconButton.propTypes = {
	id: PropTypes.string.isRequired,
	ariaLabel: PropTypes.string.isRequired,
	onClick: PropTypes.func.isRequired,
	boxShadowColor: PropTypes.string,
	unpressedBoxShadowColor: PropTypes.string,
	pressedBoxShadowColor: PropTypes.string,
	pressedBackground: PropTypes.string,
	unpressedBackground: PropTypes.string,
	pressedIconColor: PropTypes.string,
	unpressedIconColor: PropTypes.string,
	icon: PropTypes.func.isRequired,
	pressed: PropTypes.bool.isRequired,
	hoverBorderColor: PropTypes.string,
};

ChangingIconButton.defaultProps = {
	unpressedBoxShadowColor: colors.$color_button_border,
	pressedBoxShadowColor: colors.$color_purple,
	pressedBackground: colors.$color_pink_dark,
	unpressedBackground: colors.$color_button,
	pressedIconColor: colors.$color_white,
	unpressedIconColor: colors.$color_button_text,
	hoverBorderColor: colors.$color_white,
};

export default ChangingIconButton;
