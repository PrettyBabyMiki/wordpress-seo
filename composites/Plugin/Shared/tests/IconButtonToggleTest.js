import React from "react";
import renderer from "react-test-renderer";

import IconButtonToggle from "../components/IconButtonToggle";

test( "the unpressed IconButtonToggle matches the snapshot", () => {
	const component = renderer.create(
		<IconButtonToggle
			name="group1"
			id="RadioButton"
			ariaLabel="important toggle"
			icon="eye"
			pressed={ false }
			onClick={ () => {} }/>
	);

	let tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );

test( "the pressed IconButtonToggle matches the snapshot", () => {
	const component = renderer.create(
		<IconButtonToggle
			name="group1"
			id="RadioButton2"
			ariaLabel="important toggle"
			icon="eye"
			pressed={ true }
			onClick={ () => {} }/>
	);

	let tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );

test( "the disabled IconButtonToggle matches the snapshot", () => {
	const component = renderer.create(
		<IconButtonToggle
			name="group1"
			id="RadioButton2"
			ariaLabel="important toggle"
			icon="eye"
			pressed={ false }
			onClick={ () => {} }
			marksButtonStatus={ "disabled" }/>
	);

	let tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );
