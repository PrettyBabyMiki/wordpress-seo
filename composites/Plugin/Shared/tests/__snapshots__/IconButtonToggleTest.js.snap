// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the pressed IconButtonToggle matches the snapshot 1`] = `
<button
  aria-label="important toggle"
  aria-pressed={true}
  className="sc-bdVaJa VNVFd"
  id="RadioButton2"
  onClick={[Function]}
  type="button"
>
  <svg
    aria-hidden="true"
    className="sc-htpNat JvWAC"
    focusable="false"
    role="img"
  />
</button>
`;

exports[`the unpressed IconButtonToggle matches the snapshot 1`] = `
<button
  aria-label="important toggle"
  aria-pressed={false}
  className="sc-bdVaJa inADYO"
  id="RadioButton"
  onClick={[Function]}
  type="button"
>
  <svg
    aria-hidden="true"
    className="sc-bwzfXH ebHRBw"
    focusable="false"
    role="img"
  />
</button>
`;
